frontend/util/validators.js

```diff
@@ -0,0 +1,3 @@
+const validBranchName = /^[a-zA-Z0-9._-]+$/;
+
+export { validBranchName };
```

Comment:
Might want to just go ahead and make the exported value a function instead of a regex since it is used that way in both places. ie, `validBranchName = s => validBranchNameRegex.test(s)`